### ISSUE TITLE
Ignored fields not consistently exposed via `BeanDescription.getIgnoredPropertyNames()`

### ISSUE THREAD

**Issue by odrotbohm** [FasterXML/jackson-databind#3591]
The following test case fails, and I think it shouldn't:

```java
public class SampleTest {

	@Test
	void sampleTest() {

		ObjectMapper mapper = new ObjectMapper();
		SerializationConfig config = mapper.getSerializationConfig();
		ClassIntrospector introspector = config.getClassIntrospector();
		BeanDescription description = introspector.forSerialization(config, mapper.constructType(Sample.class), config);

		description.findAnySetterAccessor(); // <1>

		assertThat(description.getIgnoredPropertyNames()).containsExactlyInAnyOrder("first", "second");
	}

	@JsonIgnoreProperties("second")
	@JsonAutoDetect(fieldVisibility = Visibility.ANY)
	public class Sample {
		@JsonIgnore String first;
		String second;
	}
}
```

Note that `<1>` is even necessary to trigger the property collection in the first place. If you do not call that method, `_ignoredPropertyNames` in the underlying `POJOPropertiesCollector` will not have been initialized.

Zooming out a bit, I would like to find out about all properties, that are ignored by Jackson for either serialization or deserialization. Jackson seems to expose API to find out, but that doesn't reflect what's actually annotated in the user's sources.

**Comment by cowtowncoder** [comment]
My first thought is that this is not valid external usage: the only legit way to get `[Basic]BeanDescription` should be through `DeserializationContext` / `SerializerProvider`: other access is not supported.
Hence call to `findAnySetterAccessor()` is not required or expected: but the whole call pattern is something users should not do.

If the test can be changed to instead go through `SerializerProvider` that'd be different story so maybe test can be quickly modified that way.

But beyond that I suspect that you may be correct in that `getIgnoredPropertyNames()` does not include names defined in `@JsonIgnoreProperties` and that those are applied separately.
If so there'd be room for improvement.

**Comment by odrotbohm** [comment]
The test case simply describes a way I found to get from a type to a `BeanDescription`, which in turn seemed to offer API I was looking for. I am happy to adapt all of this the way you suggest. My core task is to be able to find out whether a particular field name is actually to be considered during serialization or deserialization (I need to be able to tell for both cases). I.e. that answer would also need to consider potential property renames (e.g. through `@JsonProperty("renamed")`).

**Comment by cowtowncoder** [comment]
Right, all I am suggesting is a change to test, I worded it badly. Plus it's not via `SerializerProvider` but `SerializationConfig`, quite similar to what you have:

```java
SerializationConfig config = mapper.getSerializationConfig();
BeanDescription desc = config.introspect(type);
```

I assume result is about the same.

And your core task sounds reasonable too so it'd be good for this to be possible.

**Comment by odrotbohm** [comment]
Result is the same indeed. Would love to see that supported. Thanks for your help already! 🙇

**Comment by cowtowncoder** [comment]
@odrotbohm I assume you are busy enough with many other things but just in case you did have time, I'd be happy to help with a PR. I can see 2 immediate possibilities;

1. Try to unify information about per-field ignorals (`@JsonIgnore`) and class-based one, so they wouldn't be separate. May be tricky, add failures
2. Add a new "getAllIgnoredPropertyNames()" convenience accessor that would combine information from 2 sources. This would not cause any incompatibilities.

of which (2) might be simpler, even if bit cumbersome to use.

**Comment by cowtowncoder** [comment]
Implemented for 3.2.0.
